docs/designs/update-strategy.md

```diff
@@ -0,0 +1,661 @@
+
+- **Enable faster rollouts** by allowing multiple replicas to be recreated simultaneously at the PCSG level for multinode deployments and PC level for singlenode inference, configurable via `maxUnavailable` settings
+
+## Non-Goals
+
+- Partition-based rolling updates or canary deployments - These are advanced features that can be considered in future phases
+
+- Automatic strategy selection based on version compatibility detection - Users must explicitly configure update strategies
+
+- Automatic selection of surge based on capacity - Users are expected to be aware of their cluster capacity and should explicitly configure surge settings. Grove does not have visibility into available cluster capacity at update time and cannot automatically determine if surge is feasible
+
+## Proposal
+
+This proposal expands Grove's default RollingUpdate strategy to support replica recreation at the PodCliqueSet level for version incompatible upgrades, as well as `maxUnavailable` and `maxSurge` configuration at the PCS level. Additionally, it introduces `maxUnavailable` and `maxSurge` concepts at the standalone PodClique and PodCliqueScalingGroup levels to enable faster rollouts and capacity maintenance during compatible upgrades.
+
+## Architecture
+
+### Three-Level Update Control
+
+```
+PodCliqueSet (Top Level)
+├─ UpdateStrategy (controls PCS replica updates)
+│  ├─ RollingUpdate: one replica at a time
+│  └─ ReplicaRecreate: tear down entire replica
+│
+├─ PCS Replica 0
+│  ├─ Standalone PodCliques (concurrent updates)
+│  │  └─ frontend: UpdateStrategy (controls pod updates)
+│  │     └─ maxUnavailable/maxSurge
+│  │
+│  └─ PodCliqueScalingGroups (concurrent updates)
+│     ├─ prefill: UpdateStrategy (controls PCSG replica updates)
+│     │  └─ maxUnavailable/maxSurge
+│     └─ decode: UpdateStrategy (controls PCSG replica updates)
+│        └─ maxUnavailable/maxSurge
+│
+└─ PCS Replica 1, 2, ... (sequential per PCS strategy)
+```
+
+## API Structure
+
+### 1. PodCliqueSetSpec.UpdateStrategy
+
+Controls **how PCS replicas are updated** (the top-most level).
+
+```go
+type PodCliqueSetUpdateStrategy struct {
+    Type          PodCliqueSetUpdateStrategyType
+    RollingUpdate *PodCliqueSetRollingUpdateStrategy
+}
+
+type PodCliqueSetUpdateStrategyType string
+const (
+    RollingUpdate    // Update replicas sequentially
+    ReplicaRecreate  // Delete/recreate entire replica
+)
+
+type PodCliqueSetRollingUpdateStrategy struct {
+    MaxUnavailable *intstr.IntOrString  // How many PCS replicas can be down (absolute number or percentage)
+    MaxSurge       *intstr.IntOrString  // How many extra PCS replicas to create (absolute number or percentage)
+}
+```
+
+**Defaults:** `RollingUpdate` with `maxUnavailable=1, maxSurge=0`
+
+### 2. PodCliqueTemplateSpec.UpdateStrategy
+
+Controls **how pods update within a standalone PodClique**.
+
+```go
+type ComponentUpdateStrategy struct {
```

Comment:
Not sure the Component is necessary here. Just call it UpdateStrategy.  Though if recreate is set on the PCS, then this doesn't matter and we probably need to warn the user.